src/core/simulation.jl

```diff
@@ -581,7 +580,10 @@ function _build!(sim::Simulation)
             throw(IS.ConflictingInputsError("Stage $(stage_name) not found in the stages definitions"))
         end
         stage_interval = get_stage_interval(sim, stage_name)
-        stage.internal.executions = Int(get_step_resolution(sim.sequence) / stage_interval)
+        step_resolution =
```

Comment:
this was only working if you had a simulation that was perfectly sequential in it's execution order. For example, if `sim.sequence.execution_order` was `[1,2,2,3,3,3,3,1,2,2,3,3,3,3]` it would work, but if it was `[1,2,3,3,2,3,3,1,2,3,3,2,3,3]` it was failing.

Reply:
We could probably make this test into a more streamlined format.